RBI-7: Logic for displaying first screen and save user's name

FarcasDavid opened the pull request:
- Created logic for displaying the screen only on the first launch and save the username on User Defaults

Commit:
RBI-7:
- Created logic for displaying the screen only on the first launch and save the username on User Defaults

edwardchristian97 commented on RecipeBook/Data/UserDefaultsName.swift:
```diff
@@ -0,0 +1,48 @@
+class UserDefaultsName {
+
+    let userDefaults = UserDefaults.standard
+
```
Remove whitespace

edwardchristian97 commented on RecipeBook/Data/UserDefaultsName.swift:
```diff
@@ -0,0 +1,48 @@
+
+    func setUserName(_ name: String) {
+        userDefaults.set(name, forKey: Keys.username.rawValue)
+
```
Remove whitespace

edwardchristian97 commented on RecipeBook/Data/UserDefaultsName.swift:
```diff
@@ -0,0 +1,48 @@
+//
+
+import Foundation
+
```
Remove whitespace

edwardchristian97 commented on RecipeBook/Data/UserDefaultsName.swift:
```diff
@@ -0,0 +1,48 @@
+        }
+    }
+
+    enum Keys: String {
```
Declare the enum outside of this class and rename it as `UserDefaultsKeys`. Also, rename this class as `UserDefaultsService`.

Commit:
RBI-7:
- Code refractoring

edwardchristian97 commented on RecipeBook/Data/UserDefaultsService.swift:
```diff
@@ -0,0 +1,45 @@
+    }
+
+    // Check first launch
+    func isFirstLaunch() -> Bool {
```
I think this logic is not necessary. It's enough to only create a getter and a setter for one key in here and from a view model use these methods to determine whether or not the app is launched for the first time.

edwardchristian97 commented on RecipeBook/Data/UserDefaultsService.swift:
```diff
@@ -0,0 +1,45 @@
+enum UserDefaultsKeys: String {
+    case username
+    case isFirstLaunch
+    case wasLaunchedBefore
```
One key is enough to check if the app is launched for the first time. Check the other comment

edwardchristian97 commented on RecipeBook/ViewControllers/HomeScreenViewController/HomeScreenViewController.swift:
```diff
@@ -18,8 +19,11 @@ class HomeScreenViewController: UIViewController {
         super.viewDidLoad()
 
         setupSearchButton()
+        let userDefaultsName = UserDefaultsService()
```
Create a view model for this class and declare `User Defaults` as a property of that class. Create methods for getting and setting the values in the view model and call them from there.

edwardchristian97 commented on RecipeBook/ViewControllers/NameScreenViewController/NameScreenViewController.swift:
```diff
@@ -19,11 +20,28 @@ class NameScreenViewController: UIViewController {
 
     func textFieldShouldReturn(_ textField: UITextField) -> Bool {
+        // Save User Credentials (Name)
+        let userDefaultsName = UserDefaultsService()
```
Create a view model for this class and declare `User Defaults` as a property of that class. Create methods for getting and setting the values in the view model and call them from there.

edwardchristian97 commented on RecipeBook/ViewControllers/NameScreenViewController/NameScreenViewController.swift:
```diff
@@ -19,11 +20,28 @@ class NameScreenViewController: UIViewController {
 
+    override func viewDidAppear(_ animated: Bool) {
+        let launchType = UserDefaultsService()
+        if !launchType.isFirstLaunch() {
```
After the view model is created, access the property within it.

Commit:
RBI-7:
- Created ViewModel for NameScreenViewController
- Created methods for handling the first launch check and saving the username in User Defaults inside the ViewModel
- Code Refractoring

edwardchristian97 commented on RecipeBook/ViewModels/NameScreenViewModel.swift:
```diff
@@ -0,0 +1,27 @@
+
+class NameScreenViewModel {
+
+    let userDefaultsProperty = UserDefaultsService()
```
Can be private

Commit:
RBI-7:
- Created ViewModel for HomeScreenViewController
- Created Method for getting the username inside HomeScreenViewModel and called it inside the ViewController

edwardchristian97 commented on RecipeBook/ViewModels/HomeScreenViewModel.swift:
```diff
@@ -0,0 +1,18 @@
+
+class HomeScreenViewModel {
+
+    let userDefaultsName = UserDefaultsService()
```
Private

Commit:
RBI-7:
- Code refractoring

edwardchristian97 commented on RecipeBook/ViewControllers/NameScreenViewController/NameScreenViewController.swift:
```diff
@@ -13,17 +14,37 @@ class NameScreenViewController: UIViewController {
     @IBOutlet private weak var nameTextField: UITextField!
 
+    // View Model
+    var viewModel: NameScreenViewModel = NameScreenViewModel()
```
Private

edwardchristian97 commented on RecipeBook/ViewControllers/HomeScreenViewController/HomeScreenViewController.swift:
```diff
@@ -14,10 +15,15 @@ class HomeScreenViewController: UIViewController {
     @IBOutlet private weak var titleMainScreenLabel: UILabel!
 
+    // ViewModel:
+    var viewModel: HomeScreenViewModel = HomeScreenViewModel()
```
Private

Commit:
RBI-7:
- Solved 2 bugs

Commit:
no message

Commit:
no message
